react-dom@2.0.3 broke strict type checking

Repeat of #2512.

This line: https://github.com/floating-ui/floating-ui/pull/2620/files#diff-d9cb4e974bdbbdd3c2c05fcfd67270b97205552f5bd0fc0e496265e497248e09R1 re-introduced a dependency on `@floating-ui/core` which breaks strict type checking:

```
@floating-ui/react-dom/src/arrow.d.ts(1,53): error TS2307: Cannot find module '@floating-ui/core' or its corresponding type declarations.
```

Reproduction is same as #2512.

--- a/.changeset/clean-ducks-tickle.md
+++ b/.changeset/clean-ducks-tickle.md
@@ -1,0 +1,5 @@
+---
+'@floating-ui/react-dom': patch
+---
+
+fix: package type import

--- a/packages/react-dom/src/arrow.ts
+++ b/packages/react-dom/src/arrow.ts
@@ -1,4 +1,4 @@
-import type {Derivable, Middleware, Padding} from '@floating-ui/core';
+import type {Derivable, Middleware, Padding} from '@floating-ui/dom';
 import {arrow as arrowCore} from '@floating-ui/dom';
 import * as React from 'react';
 

